pkg/k8s/scanner/scanner.go

```diff
@@ -143,3 +170,167 @@ func (s *Scanner) filter(ctx context.Context, r types.Report, artifact *artifact
 	}
 	return report.CreateResource(artifact, r, nil), nil
 }
+
+const (
+	golang             = "golang"
+	oci                = "oci"
+	kubelet            = "kubelet"
+	pod                = "PodInfo"
+	nodeInfo           = "NodeInfo"
+	nodeCoreComponents = "node-core-components"
+)
+
+func clusterInfoToReportResources(allArtifact []*artifacts.Artifact, clusterName string) (*core.Component, error) {
+	coreComponents := make([]*core.Component, 0)
+	for _, artifact := range allArtifact {
+		switch artifact.Kind {
+		case pod:
+			var comp bom.Component
+			err := ms.Decode(artifact.RawResource, &comp)
+			if err != nil {
+				return nil, err
+			}
+			imageComponents := make([]*core.Component, 0)
+			for _, c := range comp.Containers {
+				name := fmt.Sprintf("%s/%s", c.Registry, c.Repository)
+				cDigest := c.Digest
+				if strings.Index(c.Digest, string(digest.SHA256)) == -1 {
+					cDigest = fmt.Sprintf("%s:%s", string(digest.SHA256), cDigest)
+				}
+				imageDigest, err := cn.NewDigest(fmt.Sprintf("%s@%s", name, cDigest))
+				if err != nil {
+					return nil, xerrors.Errorf("failed to parse digest: %w", err)
+				}
+				imagePurl := purl.PackageURL{PackageURL: purl.OciPurl(imageDigest, "")}
```

Comment:
I think we can leverage the existing function.

```suggestion
				imagePURL, err := purl.NewPackageURL(purl.TypeOCI, types.Metadata{
					RepoDigests: []string{fmt.Sprintf("%s@%s", name, cDigest)},
				}, ftypes.Package{})
				if err != nil {
					return nil, xerrors.Errorf("failed to create PURL: %w", err)
				}
```